### DISCUSSION TITLE
Allow to call methods as api calls

### DISCUSSION THREAD

**Discussion by vmiguellima** [livewire/livewire#5616] [6. Polls]
Since it as been a lot of discussion around the v3 version.

I think it could be a good addition, allow to call methods like they are an api endpoint. e.g.:

```php
class EditPost extends Component
{
    #[Api]
    public preview($string body) {
      return parseMarkdown($body);
    }
}
```

And it would simply return an json response.

## Why this would be useful?

Sometimes we don't need to interact with the database and neither re-render the component. We simply want to process something on the server and show it or process it on the front end, without changing the component state:

```js
this.$wire.preview(body).then(res => this.openPreviewModal(res))
```

**Comment by joshhanley** [comment]
@vmiguellima this is already possible in V2, you just need to call `$this->skipRender()` in the method to stop Livewire from re-rendering. Any data returned from the method will be returned to the front end.

**Reply by vmiguellima** [reply]
@joshhanley I didn't know about `$this->skipRender()`. I've been looking for this for so long, thanks!

**Reply by joshhanley** [reply]
@vmiguellima no worries! 🙂 Also I've moved your discussion out of V3 as we just want that for maintainers to post in.
